### ISSUE TITLE
`CUDAExecutionProvider(...)` ctor hangs with 100% CPU in nvptxJitCompiler64

### ISSUE THREAD

**Issue by nietras** [microsoft/onnxruntime#6159]
**Describe the bug**
Creating a session with CUDA execution provider hangs during `CreateProvider()` as can be seen in call stack below.
Notice this is before any model is being parsed or used! This is on a NVidia RTX 30 series GPU, which is the only new thing. 
PTX is supposed to be forward-compatible so not sure why this hangs.

![image](https://user-images.githubusercontent.com/10798831/102486563-83170600-4069-11eb-8651-88571c0d49e5.png)
```
 	nvptxJitCompiler64.dll!00007ffadabd9a09()	Unknown
 	nvptxJitCompiler64.dll!00007ffadabd7dfa()	Unknown
 	nvptxJitCompiler64.dll!00007ffadabd5818()	Unknown
 	nvptxJitCompiler64.dll!00007ffadabd8e42()	Unknown
 	nvptxJitCompiler64.dll!00007ffadabd81e4()	Unknown
 	nvptxJitCompiler64.dll!00007ffadabf1f89()	Unknown
 	nvptxJitCompiler64.dll!00007ffadab6e8ee()	Unknown
 	nvptxJitCompiler64.dll!00007ffadab4e28d()	Unknown
 	nvptxJitCompiler64.dll!00007ffadac98952()	Unknown
 	nvptxJitCompiler64.dll!00007ffadaab1e59()	Unknown
 	nvptxJitCompiler64.dll!00007ffadaab7f4f()	Unknown
 	nvptxJitCompiler64.dll!00007ffadaab48bf()	Unknown
 	nvptxJitCompiler64.dll!00007ffadaab6b45()	Unknown
 	nvptxJitCompiler64.dll!00007ffadaabf269()	Unknown
 	nvptxJitCompiler64.dll!00007ffadaab263c()	Unknown
 	nvcuda64.dll!00007ffaca7bbadb()	Unknown
 	nvcuda64.dll!00007ffaca7bbd8b()	Unknown
 	nvcuda64.dll!00007ffaca4f81ad()	Unknown
 	nvcuda64.dll!00007ffaca4f8b53()	Unknown
 	nvcuda64.dll!00007ffaca4f949e()	Unknown
 	nvcuda64.dll!00007ffaca40a68f()	Unknown
 	nvcuda64.dll!00007ffaca40b24c()	Unknown
 	cudart64_102.dll!00007ffb21c161f3()	Unknown
 	cudart64_102.dll!00007ffb21c13569()	Unknown
 	cudart64_102.dll!00007ffb21c14908()	Unknown
 	cudart64_102.dll!00007ffb21c17a9f()	Unknown
 	cudart64_102.dll!00007ffb21c13f43()	Unknown
 	cudart64_102.dll!00007ffb21c09183()	Unknown
 	cudart64_102.dll!00007ffb21c291a0()	Unknown
>	onnxruntime.dll!onnxruntime::CUDAExecutionProvider::CUDAExecutionProvider(const onnxruntime::CUDAExecutionProviderInfo & info) Line 84	C++
 	onnxruntime.dll!std::make_unique<onnxruntime::CUDAExecutionProvider,onnxruntime::CUDAExecutionProviderInfo &,0>(onnxruntime::CUDAExecutionProviderInfo & <_Args_0>) Line 2055	C++
 	onnxruntime.dll!onnxruntime::CUDAProviderFactory::CreateProvider() Line 26	C++
 	onnxruntime.dll!`anonymous namespace'::LoadAndInitializeSession(const OrtEnv * __formal, const OrtSessionOptions * options, std::unique_ptr<onnxruntime::InferenceSession,std::default_delete<onnxruntime::InferenceSession>> & sess, OrtSession * * out) Line 470	C++
 	onnxruntime.dll!OrtApis::CreateSessionFromArray(const OrtEnv * env, const void * model_data, unsigned __int64 model_data_length, const OrtSessionOptions * options, OrtSession * * out) Line 540	C++
```

**Urgency**
Can't use onnxruntime on RTX 30 series. Can't even try to run it since it simply hangs with 100% CPU.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10 64-bit
- ONNX Runtime installed from (source or binary): Source 
- ONNX Runtime version: 1.0.0.0
- Python version: Ugh
- Visual Studio version (if applicable): 2019 16.8
- GCC/Compiler version (if compiling from source): MSVCC
- CUDA/cuDNN version: 10.2/7
- GPU model and memory: RTX 3070 8 GB

**To Reproduce**
Simply try creating a session with CUDA/TensorRT provider.

**Expected behavior**
Provider can be created. PTX is forward-compatible. Old CUDA versions can run on newer GPUs if used correctly.

**Comment by faxu** [comment]
@nietras are you still experiencing this issue with the latest version of ORT?

**Comment by nietras** [comment]
@faxu no I solved this by compiling with CUDA 11.2 and TensorRT 7.2 that supports Ampere. Thanks.
